docs/platforms/javascript/common/troubleshooting/initialization-methods.mdx

```diff
@@ -0,0 +1,70 @@
+---
+title: Initialization Methods
+description: "Review our alternate ways to initialize and run Sentry."
+supported:
+  - javascript.node
+  - javascript.koa
+  - javascript.connect
+  - javascript.fastify
+  - javascript.express
+---
+
+The Sentry Node.js SDK can be initialized in multiple ways. Read this page to figure out which one to use.
+
+## How To Decide Which Method To Use
+
+You can initialize the SDK by either
+
+- Using the `--require` (CJS) or `--import` (ESM) flags to import a dedicated file to initialize the SDK (recommended)
+- Importing a dedicated file to initialize Sentry from within your application
+
+### `--require` Or `--import`
+
+Using [--require](https://nodejs.org/api/cli.html#-r---require-module) or [--import](https://nodejs.org/api/cli.html#--importmodule) guarantees that Sentry is imported and initialized before any other modules in your application, which is necessary for the SDK to work properly.
+
+```bash
+# If you are using CommonJS (CJS)
+node --require ./instrument.js app.js
+
+# If you are using ECMAScript Modules (ESM)
+# Note: This is only available for Node v18.19.0 onwards.
+node --import ./instrument.mjs app.mjs
+```
+
+In some situations however, it may not be possible for you to use these flags:
+
+- **No access to Node.js arguments**: There are many ways of running Node.js and it may not be possible for you to change the runtime flags due to technical or organizational limitations.
+- **Old Node.js versions**: If you are using ECMAScript modules (ESM), you must load your dedicated Sentry initialization file with the `--import` flag. This flag was only introduced in Node.js v18.18.0 and onwards.
+
+If you find yourself in one of these situations, you can evaluate simply importing your Sentry initialization file with an import statement inside your application instead.
+
+### Importing The Sentry Initialization File Directly
+
+As an alternative to the `--require` or `--import` flags, you can simply import your Sentry initialization file directly from your application.
+**Please note that this is not recommended** since `Sentry.init()` should be run before any `import` or `require` statements.
+The Sentry SDK depends on import ordering for instrumenting native and third-party modules.
+We discourage directly importing the initialization file since import ordering is error-prone or can be flagged and mistakenly auto-fixed by linters or changed by code formatters.
+
+```javascript {tabTitle:CommonJS}
+// Ensure to require this before requiring any other modules!
+require("./instrument.js");
+
+// http and other libraries are only instrumented
+// if required _after_ Sentry has been initialized
+const http = require("http");
+```
+
+```javascript {tabTitle:ESM}
+// Ensure to import this before importing any other modules!
+import "./instrument.js";
+
+// http and other libraries are only instrumented
+// if imported _after_ Sentry has been initialized
+import http from "http";
+```
```

Comment:
If node is running ESM source, ie. you're not transpiling the above to CJS, my understanding is that import ordering makes zero difference. `import http from "http"` is evaluated and loaded before any code in `instrument.js` is even executed and libraries will not be instrumented without the loader detailed below anyway. 

When you are using the loader, `import-in-the-middle` wraps everything in proxies to work around the whole "imports evaluated first" problem.

Of course, users might be transpiling to CJS, even without their knowledge so maybe it's still important to keep the ordering, but then the loader below will not be required anyway!